plugins/nf-co2footprint/src/main/nextflow/co2footprint/CO2FootprintFactory.groovy

```diff
@@ -114,8 +115,23 @@ class CO2FootprintFactory implements TraceObserverFactory {
     }
 
 
+    private String convertToReadableUnits(double value, int unitIndex=4) {
+        def units = ['p', 'n', 'u', 'm', ' ', 'K', 'M', 'G', 'T', 'P', 'E']  // Units: pico, nano, micro, mili, 0, Kilo, Mega, Giga, Tera, Peta, Exa
```

Comment:
like this, if the unit is `gram` no unit is given out. Shouldn't `g` be given out as well?

Reply:
ah, my mistake :D it's just the prefix

Reply:
This function outputs only the "prefix", the actual unit should be added afterwards, that way we can reuse it for different units, such as g, Wh...

Reply:
exactly, I didn't see your second message before replying 😄